### PR TITLE
[iosapp] Refactor preparations

### PR DESCRIPTION
**Pull request by captainbarbosa**
This change does the following:
- Includes https://github.com/mapbox/ios-sdk-examples/ as a submodule of this project, located at `platform/ios/ios-sdk-examples/`
- Removes functionality that already exists as an example within `ios-sdk-examples`

### CODE REVIEW

**Comment by captainbarbosa** [comment]
Per chat with @jmkiley additional features and refactoring will be PR'd to this branch. Here is the task list for remaining things we'd like to do:

- [ ] Break out state saving into different files (PR → )
- [ ] Add reset/clear button as a navigation menu bar item (PR → )
- [ ] Break out debug settings into a separate file (PR → )
- [ ] Move the settings table view controller into a separate file (PR → )
- [ ] Move removed features into `ios-sdk-examples` (PR → )

File: platform/ios/ios.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1447,6 +1420,14 @@
 /* End PBXFrameworksBuildPhase section */
 
 /* Begin PBXGroup section */
+		074A7F062277792F001A62D1 /* Recovered References */ = {
```
Comment (friedbunny) [inline platform/ios/ios.xcodeproj/project.pbxproj]: This shouldn’t exist — we’ll want to resolve this issue before merging.

File: platform/ios/ios.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2958,6 +2917,28 @@
...
+				"$(TARGET_BUILD_DIR)/$(INFOPLIST_PATH)",
+			);
+			name = "Insert Mapbox Access Token";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "if [[ \"$CI\" ]]; then\n  echo \"CI environment, access token not required\"\n  exit 0\nelse\n  token_file=$SRCROOT/mapbox_access_token\n  token=\"$(cat $token_file 2> /dev/null)\" || token=\"$(cat ~/.mapbox 2> /dev/null)\"\n\n  if [ \"$token\" ]; then\n    plutil -replace MGLMapboxAccessToken -string $token \"$TARGET_BUILD_DIR/$INFOPLIST_PATH\"\n  else\n    echo 'error: Missing Mapbox access token'\n    echo \"error: Get an access token from <https://www.mapbox.com/studio/account/tokens/>, then create a new file at $token_file that contains the access token.\"\n  exit 1\nfi\n\nfi\n\n";
```
Comment (friedbunny) [inline platform/ios/ios.xcodeproj/project.pbxproj]: There are some tradeoffs to directly including scripts in projects — they’re a bit difficult to read/review and can cause merge conflicts, but it’s perhaps a more intuitive editing experience when they’re in Xcode.

The alternative is to break this out into its own file, then invoke that file here in the project. Separate files are better formatted (in pull requests, at least) and able to be run independently, so I tend to prefer them.

File: platform/ios/app/MBXViewController.m [hunk 3]
```diff
@@ -1954,58 +1004,6 @@ - (IBAction)handleLongPress:(UILongPressGestureRecognizer *)longPress
     }
 }
 
-- (IBAction)cycleStyles:(__unused id)sender
```
Comment (friedbunny) [inline platform/ios/app/MBXViewController.m]: What’s your plan for including this functionality going forward?

**Comment by friedbunny** [comment]
Separately, is `ios-sdk-examples` ready to be brought into this project? That repo contains a fair bit of cruft not directly related to the maps SDK (e.g., navigation examples, swiftlint, cocoapods, etc.) which could make integration here difficult/inefficient.

File: platform/ios/missing_icon.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
+// 20190423145312
```
Comment (friedbunny) [inline platform/ios/missing_icon.json]: I know you already removed this junk, but the file itself should also live in a more specific path.

**Comment by captainbarbosa** [comment]
After talking with @friedbunny, it seems like this PR is best split up into multiple to separate concerns 😄

**Pull request closed without merging**